### PR TITLE
fix(ecs): unclear docs on valid ec2 task definition placement constraints and no validation of input #25933

### PR DESCRIPTION
**Pull request by Adibuer-lab**
Docs were unclear on valid placement constraints. You can specify distinctInstances when creating a Service, or running a task. You can specify memberOf when doing the previous two actions, or when creating a task or new revision. So far only memberOf is valid for ec2 task definitions.

This pull request enhances the documentation and validation for task definition placement constraints in the aws-cdk-lib/aws-ecs package. The documentation now includes a note clarifying the valid placement constraints and a new validatePlacementConstraints method has been added to the Ec2TaskDefinition class to validate the constraints at synth time.

The changes include:

Enhancement of documentation in ec2-task-definition.ts to include a note about valid placement constraints.
Addition of the validatePlacementConstraints method in ec2-task-definition.ts to validate placement constraints at synth time.
Addition of unit tests in ec2-task-definition.test.ts to verify the correct behavior of the new validatePlacementConstraints method.
Addition of integration tests to ensure overall functionality.
Closes https://github.com/aws/aws-cdk/issues/25905.

Revival of #25933

----

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license

### CODE REVIEW

**Comment by Adibuer-lab** [comment]
@kaizencc made requested changes

**Comment by mrgrain** [comment]
Hi and thanks for your contribution! We can only accept contributions that comply with our license requirements.
Please include the following statement in you PR description.

```
----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*
```

**Comment by Adibuer-lab** [comment]
> Hi and thanks for your contribution! We can only accept contributions that comply with our license requirements. Please include the following statement in you PR description.
> 
> ```
> ----
> 
> *By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*
> ```

Done :)

**Review by kaizencc — CHANGES_REQUESTED** [review]
Hi @Adibuer-lab! Thanks for reviving this PR! Just one comment about the valid constraints, did one just get added?

File: packages/aws-cdk-lib/aws-ecs/lib/ec2/ec2-task-definition.ts [hunk 1]
```diff
@@ -110,6 +109,26 @@ export class Ec2TaskDefinition extends TaskDefinition implements IEc2TaskDefinit
     });
   }
 
+  /**
+   * Validates the placement constraints to make sure they are supported.
+   * Currently, only 'memberOf' is a valid constraint for an Ec2TaskDefinition.
+   */
+  private static validatePlacementConstraints(constraints?: PlacementConstraint[]) {
+    // List of valid constraints https://docs.aws.amazon.com/AmazonECS/latest/developerguide/task-placement-constraints.html
```
[the remark this answers is not in the document]
Comment (Adibuer-lab) [reply packages/aws-cdk-lib/aws-ecs/lib/ec2/ec2-task-definition.ts:118]: Ah that's where the confusion arises, I should've used the [cloudformation docs for task definition](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-ecs-taskdefinition-taskdefinitionplacementconstraint.html)  instead.According to the docs You can specify distinctInstances when creating a Service, or running a task but not when creating a task definition- cloudformation docs also show only accepting memberOf. And according to [here](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/task-placement-constraints.html) as well, distinctInstance can be used when either running a task or creating a new service but doesn't mention creating a task definition. So doesn't look like docs have changed, will try reproducing the error again and see what happens.

File: packages/aws-cdk-lib/aws-ecs/lib/ec2/ec2-task-definition.ts [hunk 2]
```diff
@@ -110,6 +110,26 @@ export class Ec2TaskDefinition extends TaskDefinition implements IEc2TaskDefinit
...
```
Comment (kaizencc) [inline packages/aws-cdk-lib/aws-ecs/lib/ec2/ec2-task-definition.ts:118]:
```suggestion
    // List of valid constraints https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-ecs-taskdefinition-taskdefinitionplacementconstraint.html#cfn-ecs-taskdefinition-taskdefinitionplacementconstraint-type
```

**Review by kaizencc — APPROVED** [review]
Thanks @Adibuer-lab

**Pull request merged**